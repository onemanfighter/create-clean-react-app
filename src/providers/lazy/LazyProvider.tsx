import * as React from 'react';
import { LazyProviderProps } from './types';
import { LoadingComponent } from '@components';

const LazyProvider = ({ children }: LazyProviderProps) => {
  return (
    <React.Suspense fallback={<LoadingComponent />}>{children}</React.Suspense>
  );
};

export default LazyProvider;
