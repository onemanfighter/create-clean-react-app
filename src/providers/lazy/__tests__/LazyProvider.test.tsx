import { render, screen } from '@testing-library/react';
import LazyProvider from '../LazyProvider';
import { Text } from '@chakra-ui/react';

describe('LazyProvider', () => {
  it('should render correctly', () => {
    const { container } = render(
      <LazyProvider>
        <Text>Test</Text>
      </LazyProvider>,
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText('Test')).toBeDefined();
  });
});
