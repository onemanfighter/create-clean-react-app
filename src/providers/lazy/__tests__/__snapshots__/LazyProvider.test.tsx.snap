// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyProvider should render correctly 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    Test
  </p>
</div>
`;
